### PR TITLE
Research Order of the Sun ugprades

### PR DESCRIPTION
**Pull request by cokernel**
This is a partial implementation of #149.  I'm not convinced this is ready to merge, but it's easier to reference this branch if there's an active pull request.

Right now, I'm only handling Order of the Sun upgrades.  During a "worship" phase, the Religion manager tries to research each of the upgrades in order (correctly handling the case where Transcendence is unlocked), then uses any unspent faith to praise the Sun.  It does not research multiples of an upgrade, even if it is possible to do so.  By default, Apocrypha will not be researched, since it usually doesn't make sense to use it until a reset.

The Religion manager in Kittens Game differs from the Building manager in that the buttons are arranged into `zgUpgradeButtons` and `rUpgradeButtons` instead of just `buttons`.  Because I'm following the Building manager in Kitten Scientists quite closely for my implementation, I decided to skip Ziggurats for this first version.  I think it should be relatively straightforward, however, to modify `ReligionManager.getBuild()` and `ReligionManager.getBuildButton()` to check both button lists.  Of course, getting the Ziggurat path right would be more involved than that, as it would essentially require implementing the Unicorn worksheet discussed in #144.

### CODE REVIEW

**Comment by oliversalzburg** [comment]
Neat. I'll test this on my next reset.

**Comment by amaranth** [comment]
This makes the Religion tab have the same bug with buttons not updating as the Bonfire tab in the latest version but otherwise seems to work great. It's been automatically buying religion upgrades as they're available and I like that it kind of evenly upgrades them by not doing the first one as many times as possible.

File: kitten-scientists.user.js [hunk 1]
```diff
@@ -358,14 +374,34 @@ Engine.prototype = {
...
+    worship: function () {
+        var builds = options.auto.faith.items;
+        var buildManager = this.religionManager;
+        var craftManager = this.craftManager;
+        var trigger = options.auto.faith.trigger;
+
+        // Render the tab to make sure that the buttons actually exist in the DOM. Otherwise we can't click them.
+        buildManager.manager.render();
+
+        for (var name in builds) {
```
Comment (woutershep) [inline kitten-scientists.user.js]: This loop is missing `if (!builds[name].enabled) continue;`.

**Comment by cameroncondry** [comment]
Going to close this PR for now since it is no longer relevant. If anyone wants to update this fully, then feel free and I'll pull it in.

**Pull request closed without merging**